test/unit/puppetlabs/services/jruby_pool_manager/jruby_ref_pool_test.clj

```diff
@@ -118,6 +125,40 @@
           (is (= id (:id instance)))
           (jruby-core/return-to-pool pool-context instance :test []))))))
 
+(deftest flush-times-out-on-return
+  (testing "Attempt to flush times out if flush-timeout is reached"
+    (jruby-testutils/with-pool-context
+      pool-context
+      jruby-testutils/default-services
+      ;; Set max-instances higher than max-borrows in order to over-borrow
+      (jruby-test-config 1 2 {:flush-timeout 1})
+      ;; Borrow instance1 so we can trigger a refresh when we return it
+      ;; Borrow instance2 to prevent lock from being acquired when we return instance1
+      (let [instance1 (pool-protocol/borrow pool-context)
+            instance2 (pool-protocol/borrow pool-context)
+            flush-complete (add-watch-for-flush-complete pool-context)]
+        ;; Return to trigger flush, which should lock since instance2 is still borrowed
+        (pool-protocol/return pool-context instance1)
+        (is (not (realized? flush-complete)))
+        ;; Return instance2 so that shutdown can proceed (otherwise this hangs)
```

Comment:
I think this comment might be misleading now, since we're blocking on checking for the flush to complete below. It should probably reference the flush instead of shutdown.

Reply:
hmm I'm starting to wonder if this is really testing what we want to test.. 
I feel like this isn't proving that the flush timed out.. in theory it could just be waiting, right?

Reply:
That's a good point. I wonder if we can check for logging...?